test_pull_requests

```diff
@@ -935,6 +1023,103 @@ popd
+
+      comments = sort_comments(comments)
+      comments.each do |comment|
+        comment_matches_syntax = comment['body'] =~ /#{Regexp.quote(build_url)}/ && comment['body'] =~ general_issue_spec
+        # we can't allow the bot to be a valid author, or we would pick up
+        # comments where the bot explains the flake syntax to GitHub users
+        comment_author_valid = comment['user']['login'] != Properties['bot_github_user']
+        if comment_matches_syntax && comment_author_valid
+          # if we find a comment that matches the flake job syntax from someone
+          # trusted, we need to validate the comment's flake issue list
+          bad_issue_links = []
+          comment['body'].scan(general_issue_spec) do |issue|
+            if issue =~ flake_issue_spec
+              issue_url = $1
+              issue_id = $2.to_i
+              if !issue_has_label?(issue_id, flake_repo, flake_label)
+                # the issue does not have the labels necessary, so it is invalid
+                bad_issue_links << issue_url
+              else
+                valid_comment_exists = true
+              end
+            else
+              # we had a GitHub issue link, but it didn't
+              # match the org or repo that we wanted
+              bad_issue_links << issue
+            end
+          end
+
+          if valid_comment_exists
+            break
+          else
+            # no valid issue links mean that the comment is invalid
+            reason = %{A [comment](#{comment["html_url"]}) was found that referenced the correct Jenkins job URL, but no listed issue links were valid. Ensure at least one linked issue is in the [`#{org_repo}` repository](#{issue_link}) and has the `#{flake_label}` label. Invalid links are:
+ - #{bad_issue_links.join("\n - ")}}
+          end
+        end
+      end
+
+      [valid_comment_exists, reason]
+    end
+
+    # issue_has_label determines if the issue at the
+    # repo and issue ID has a label with the given name
+    def issue_has_label?(issue_id, repo, label)
+      get_labels(issue_id, repo).each do |label|
+        if label["name"] == label
+          return true
+        end
+      end
+      return false
+    end
+
+    # format_teams builds a list of links to team rosters from a list of team IDs
+    def format_teams(team_ids)
+      links = []
+      team_ids.each do |id|
+        url, name = get_team_url_and_name(id)
+        links << "[#{name}](#{url})"
+      end
+      if links.length == 1
+        %{the #{links[0]} group}
+      else
+        %{the following groups: #{links.join(", ")}}
+      end
+    end
+
+    # format_flake_comment builds a comment explaining why a re-build could not be triggered
+    def format_flake_comment(repo, job, reason, admin_groups)
```

Comment:
Shouldn't pass a formatted string to a format message.  It makes assumptions about what the final format should be.